com.unity.render-pipelines.core/Editor/ContextualMenuDispatcher.cs

```diff
@@ -54,99 +54,91 @@ static void RemoveCameraComponent(MenuCommand command)
         {
             T comp = command.context as T;
-
-            if (!DispatchRemoveComponent<T>(comp))
-            {
-                //preserve built-in behavior
-                if (RemoveComponentUtils.CanRemoveComponent(comp, RemoveComponentUtils.ComponentDependencies(comp)))
-                    Undo.DestroyObjectImmediate(command.context);
-            }
+            RemoveComponent(typeof(T), comp);
         }
 
-        static bool DispatchRemoveComponent<T>(T component)
-            where T : Component
+        /// <summary>
+        /// Removes the component and it's additional datas
+        /// </summary>
+        /// <param name="type">Type to remove</param>
+        /// <param name="comp">The component</param>
+        public static void RemoveComponent(Type type, Component comp)
         {
-            try
+            if (!RemoveAdditionalDataUtils.RemoveComponent(type, comp, RemoveComponentUtils.ComponentDependencies(comp)))
             {
-                var instance = new RemoveAdditionalDataContextualMenu<T>();
-                instance.RemoveComponent(component, RemoveComponentUtils.ComponentDependencies(component));
-                return true;
-            }
-            catch
-            {
-                return false;
+                //preserve built-in behavior
+                if (RemoveComponentUtils.CanRemoveComponent(comp, RemoveComponentUtils.ComponentDependencies(comp)))
+                    Undo.DestroyObjectImmediate(comp);
             }
         }
     }
 
-    /// <summary>
-    /// Interface that should be used with [ScriptableRenderPipelineExtension(type))] attribute to dispatch ContextualMenu calls on the different SRPs
-    /// </summary>
-    /// <typeparam name="T">This must be a component that require AdditionalData in your SRP</typeparam>
-    [Obsolete("The menu items are handled automatically for components with the AdditionalComponentData attribute", false)]
-    public interface IRemoveAdditionalDataContextualMenu<T>
-        where T : Component
+    static class RemoveAdditionalDataUtils
     {
         /// <summary>
         /// Remove the given component
         /// </summary>
+        /// <param name="type">The type of the component</param>
         /// <param name="component">The component to remove</param>
         /// <param name="dependencies">Dependencies.</param>
-        void RemoveComponent(T component, IEnumerable<Component> dependencies);
-    }
-
-    internal class RemoveAdditionalDataContextualMenu<T>
-        where T : Component
-    {
-        /// <summary>
-        /// Remove the given component
-        /// </summary>
-        /// <param name="component">The component to remove</param>
-        /// <param name="dependencies">Dependencies.</param>
-        public void RemoveComponent(T component, IEnumerable<Component> dependencies)
+        public static bool RemoveComponent(Type type, Component component, IEnumerable<Component> dependencies)
```

Comment:
I think `type` is redundant, have you considered deducing `type` from `component`?

```suggestion
        public static bool RemoveComponent([DisallowNull] Component component, IEnumerable<Component> dependencies)
```